exchanges/gateio/gateio_websocket_test.go

```diff
@@ -0,0 +1,176 @@
+package gateio
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/account"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+)
+
+func TestProcessSpotBalances(t *testing.T) {
+	t.Parallel()
+	e := new(Exchange) //nolint:govet // Intentional shadow
+	e.SetDefaults()
+	e.Name = "ProcessSpotBalancesTest"
+
+	ctx := account.DeployCredentialsToContext(t.Context(), &account.Credentials{Key: "test", Secret: "test"})
+
+	// t.Parallel() not required below as this is sequenced. Some timestamps are identical from trading activity.
+	for _, tc := range []struct {
+		input    []byte
+		err      error
+		ctx      context.Context
+		expected []account.Change
```

Comment:
🚧 Can we break this type out, since we use it twice.
And, then ... we can use it thrice ... 😈 in the checker

Reply:
yep good suggestion